keyboards/kbd6x/keymaps/othi/keymap.c

```diff
@@ -0,0 +1,225 @@
+                return false;
+        }
+    }
+    return true;
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
+    switch (biton32(state)) {
+    case NM_MODE:
+        rgblight_setrgb (0x00,  0x66, 0x00);
+        break;
+    case VI_MODE:
+        rgblight_setrgb (0x66,  0x66, 0x00);
+        break;
+    case ACCENT:
+        rgblight_setrgb (0x7A,  0x00, 0xFF);
+        break;
+    case ACCENT_CAP:
+        rgblight_setrgb (0x7A,  0xFF, 0xFF);
+        break;
+    default: //  for any other layers, or the default layer
+        rgblight_setrgb (0xFF,  0x00, 0x00);
+        break;
+    }
+  return state;
+}
+
+//Tap Dance Declarations
+enum {
+  CTRL_NM = 0,
+  ALT_NM = 1,
+  SFT_NM = 2,
+  GUI_NM = 3
+};
+
+void dance_CTRL_NM_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+	register_code (KC_LCTL);
+	set_oneshot_mods(MOD_LCTL);
+  } else {
+    register_code (KC_LCTL);
+	layer_on(NM_MODE);
+  }
+}
+
+void dance_CTRL_NM_reset (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    unregister_code (KC_LCTL);
+  } else {
+    unregister_code (KC_LCTL);
+	layer_off(NM_MODE);
+  }
+}
+
+void dance_GUI_NM_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+	register_code (KC_LGUI);
+  } else {
+    register_code (KC_LGUI);
+	layer_on(NM_MODE);
+  }
+}
+
+void dance_GUI_NM_reset (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    unregister_code (KC_LGUI);
+  } else {
+    unregister_code (KC_LGUI);
+	layer_off(NM_MODE);
+  }
+}
+
+void dance_ALT_NM_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+	register_code (KC_LALT);
+  } else {
+    register_code (KC_LALT);
+	layer_on(NM_MODE);
+  }
+}
+
+void dance_ALT_NM_reset (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    unregister_code (KC_LALT);
+  } else {
+    unregister_code (KC_LALT);
+	layer_off(NM_MODE);
+  }
+}
+
+void dance_SFT_NM_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+	register_code (KC_LSFT);
+	set_oneshot_mods(MOD_LSFT);
+  } else {
+    register_code (KC_LSFT);
+	layer_on(NM_MODE);
+  }
+}
+
+void dance_SFT_NM_reset (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    unregister_code (KC_LSFT);
+  } else {
+    unregister_code (KC_LSFT);
+	layer_off(NM_MODE);
+  }
+}
+
+
+qk_tap_dance_action_t tap_dance_actions[] = {
+ [CTRL_NM] = ACTION_TAP_DANCE_FN_ADVANCED (NULL, dance_CTRL_NM_finished, dance_CTRL_NM_reset),
+ [GUI_NM] = ACTION_TAP_DANCE_FN_ADVANCED (NULL, dance_GUI_NM_finished, dance_GUI_NM_reset),
+ [ALT_NM] = ACTION_TAP_DANCE_FN_ADVANCED (NULL, dance_ALT_NM_finished, dance_ALT_NM_reset),
+ [SFT_NM] = ACTION_TAP_DANCE_FN_ADVANCED (NULL, dance_SFT_NM_finished, dance_SFT_NM_reset)
+};
+
+// Fillers to make layering clearer
```

Comment:
QMK defines a keycode for this automatically.

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L29-L31

If you apply the attached diff, you won't need Lines 182-183. (Attached a diff because you actually have a couple instances of the seven-character core keycode.)

[kbd6x_othi.diff.txt](https://github.com/qmk/qmk_firmware/files/2704716/kbd6x_othi.diff.txt)